server/semantics-service/src/main/java/edu/illinois/ncsa/incore/service/semantics/utils/CommonUtil.java

```diff
@@ -0,0 +1,20 @@
+package edu.illinois.ncsa.incore.service.semantics.utils;
+
+import edu.illinois.ncsa.incore.service.semantics.model.Type;
+import java.util.Comparator;
+
+public class CommonUtil {
+    public static Comparator<Type> typeComparator(String sortBy, String order){
+        // construct comparator
+        Comparator<Type> comparator;
+        if (sortBy.equals("name")) {
+            comparator = Comparator.comparing(Type::getTitle);
+        }
+        else{
+            // default to id
+            comparator = Comparator.comparing(Type::getId);
```

Comment:
Sorry Last thing, i noticed sortBy is actually not being used unless it is "name". Should we port this in? if sortBy is null then default to ID?

Reply:
You are right. Maybe we can stick to this pattern and think about it later :-)